src/component/tooltip/tooltipMarkup.ts

```diff
@@ -480,9 +480,8 @@ export function retrieveVisualColorForTooltipMarker(
 
 export function retrieveVisualOpacityForTooltipMarker(
     series: SeriesModel,
-    dataIndex: number
 ): number {
-    const style = series.getData().getItemVisual(dataIndex, 'style');
+    const style = series.getData().getVisual('style');
```

Comment:
This may not be correct. With this change, the opacity in the tooltip of pie series cannot use `series.data.itemStyle.opacity`.

Reply:
<img width="964" alt="image" src="https://github.com/apache/echarts/assets/56526981/9ca2b827-b3d5-448a-a910-2d85fc7fa042">

After some testing, I found that the brush operation changes the colour of the graphic item, which will result in a colour change when tooltip is used, so would it also make sense for the brush operation to change the opacity of the tooltip?

Reply:
I don't think so. The color should remain the same just as `emphasis` color should not be used when hovered.